staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -1407,3 +1409,193 @@ func isStreamReset(err error) bool {
 	}
 	return false
 }
+
+func TestGetRequestWaitContextWithFake(t *testing.T) {
+	tests := []struct {
+		name                    string
+		defaultRequestWaitLimit time.Duration
+		parent                  func(time.Time) (context.Context, context.CancelFunc)
+		isWatch                 bool
+		reqWaitCtxExpected      bool
+		reqWaitLimitExpected    time.Duration
+	}{
+		{
+			name: "parent context has already expired",
+			parent: func(time.Time) (context.Context, context.CancelFunc) {
+				return &fakeContext{err: context.Canceled}, func() {}
+			},
+			reqWaitCtxExpected: false,
+		},
+		{
+			name:                    "request is a watch",
```

Comment:
This is not the only case for which "request is a watch" is an accurate description, more detail here would be helfpul.